[24] Refactor filters class

carlosdanielpohlod opened the pull request:
Issue: [refactor filter class](https://github.com/carlosdanielpohlod/rails_tracepoint_stack/issues/24)

Changes:

 Refactored the trace filters in 4 different classes:

`CustomTraceSelectorFilter`:  filters a trace by the `file_path_to_filter_patterns` configuration attribute

`TraceFromDependenciesFilter`: filters a trace by the trace path, checking if is not part of a gem, a ruby lib path, or a bundler dependency

`TraceFromRubyCodeFilter`: filters a trace analysing if the trace is not related to an eval or internal

`TraceToIgnoreFilter`: filters considering if the trace path is not contained on the configuration `ignore_patterns` field

carlosdanielpohlod committed:
fix gemfile gem version

carlosdanielpohlod committed:
fix bool condition on CustomTraceSelectorFilter

carlosdanielpohlod committed:
tests: applied some adjusts on CustomTraceSelectorFilter and add tests

carlosdanielpohlod committed:
refactor:

some fixes on the filters and improve the tracer test coverage

carlosdanielpohlod committed:
fix: remove byebug

carlosdanielpohlod committed:
tests: applied some fixes

carlosdanielpohlod committed:
removed some prints

carlosdanielpohlod committed:
fix: fix if sentence on filter

carlosdanielpohlod committed:
remove print

fabiosammy commented on spec/spec_helper.rb:
```diff
@@ -8,4 +8,15 @@ def initialize_gem_configuration!
+    RailsTracepointStack.configuration = original_configuration
+  end
+
+  # Desabilita o tracer globalmente nos testes
```
english please

fabiosammy commented on spec/rails_tracepoint_stack/tracer_spec.rb:
```diff
@@ -40,6 +40,7 @@ def dummy_method_with_params(param_1, param_2)
       before do
         RailsTracepointStack.configure do |config|
           config.log_format = :text
+          config.log_external_sources = false
```
this is false by default, no?

fabiosammy commented on lib/rails_tracepoint_stack/filter/custom_trace_selector_filter.rb:
```diff
@@ -0,0 +1,19 @@
+module RailsTracepointStack
+  module Filter
+    class CustomTraceSelectorFilter
+      def self.ignore_trace?(trace:)
```
maybe it is better to create modules and include them in the main class instead of passing the trace as a parameter to each class method. 
The current approach on this PR is not a great OO example.

What do you think?

carlosdanielpohlod committed:
change CustomTraceSelectorFilter to be a module

carlosdanielpohlod committed:
change TraceFromDependenciesFilter to be a module

carlosdanielpohlod committed:
refactor: change CustomTraceSelectorFilter to be a module instead a class

carlosdanielpohlod committed:
refactor: change TraceToIgnoreFilter to be a module and rename the main method to attends_some_custom_pattern_to_ignore

carlosdanielpohlod committed:
refactor:

change all filters and the TraceFilter class to be a module without self methods

carlosdanielpohlod committed:
fix tests:

add include of each module on filters tests

carlosdanielpohlod committed:
fix: remove unused changes on spec_helper

carlosdanielpohlod committed:
chore: remove unused setup on test
